internal/rbd/replicationcontrollerserver.go

```diff
@@ -420,6 +511,15 @@ func (rs *ReplicationServer) PromoteVolume(ctx context.Context,
 
 	// promote secondary to primary
 	if !mirroringInfo.Primary {
+		var mode librbd.ImageMirrorMode
+		mode, err = getMirroringMode(ctx, req.GetParameters())
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, "failed to get mirroring node %s", err.Error())
+		}
+		err = enableMirroringOnDummyImage(rbdVol, mode)
```

Comment:
rbdVol is not the dummy volume.  We will be enabling mirroring on a none dummy volume if am reading this correctly.

Reply:
Also, enabling mirroring for the dummy volume should be done after we promote the real volume.

Reply:
`enableMirroringOnDummyImage` takes the rbdVol as input and creates a new dummy struct and replaces the name and calls the mirror operation on that dummy image.

Reply:
@BenamarMk is it the same for disabling mirroring on dummy image also? it should be done after demoting the real image?

Reply:
fixed for enabling. let me know your input for disable also.

Reply:
Disable is fine.  YOu don't need to do anything to it.  So basically the flow is like this:
1. Promote
2. enable dummy
And then
1. Disable dummy
2. Demote

Reply:
okay, then the current changes are fine. Thanks for confirming @BenamarMk